dev/create-release/release-tag.sh

```diff
@@ -106,6 +106,8 @@ sed -i".tmp7" 's/SPARK_VERSION:.*$/SPARK_VERSION: '"$NEXT_VERSION"'/g' docs/_con
 sed -i".tmp8" 's/SPARK_VERSION_SHORT:.*$/SPARK_VERSION_SHORT: '"$R_NEXT_VERSION"'/g' docs/_config.yml
 # Update the version index of DocSearch as the short version
 sed -i".tmp9" "s/'facetFilters':.*$/'facetFilters': [\"version:$R_NEXT_VERSION\"]/g" docs/_config.yml
+# Remove test jars that do not beling to source releases.
+rm $(<dev/test-jars.txt)
```

Comment:
Remove jars, commit, then push the change to remote? 

What are the differences between it and deleting jars in this PR directly?

Reply:
In this way, the tagged commits (source releases) would not have the jars.

Reply:
but the tag also belongs to the release branch, e.g. branch-4.0, then all commits after this tag do not have the jars, and it affects code repo too.

Currently, Spark clones the repo and uses shell commands like `rm`, `tar`, to create the source release tarball, 
https://github.com/apache/spark/blob/3ffb8009a3d660f25b4d003d0e702d60e8527ab6/dev/create-release/release-build.sh#L263

I think the simplest way is to replace it with `git archive`, with a `.gitattributes` to define the excluded files, e.g.
https://github.com/apache/kyuubi/blob/176bc293fc142b3b9bf07dddd40e525072905efb/build/release/create-package.sh#L61
https://github.com/apache/kyuubi/blob/176bc293fc142b3b9bf07dddd40e525072905efb/.gitattributes#L25

Reply:
> ... tagged commits (source releases) ...

According to https://infra.apache.org/release-publishing.html, the Apache source release is something like https://dist.apache.org/repos/dist/release/spark/spark-3.5.5/spark-3.5.5.tgz, not https://github.com/apache/spark/tree/v3.5.5

Reply:
This is a great point. I'll take a deeper look.

Reply:
Yeah, so this does include those jars. And, with this change, they will be excluded.

Reply:
@HyukjinKwon Suppose we are releasing version 4.0. Is the process for `tag` after this pull request as follows?

1. Pull a `branch-4.0` branch.
2. Remove jars from `branch-4.0`.
3. Set the version to 4.0.0 using Maven.
4. Use this branch to create a `4.0-rcX` tag. At this point, the `4.0-rcX` tag will not include these jars.
5. Set the version to 4.0.1-SNAPSHOT using Maven.
6. Push the previous commits to `branch-4.0`.

If this is the case, will the jars removed in step 2 be permanently lost after a `tag` operation? It seems that we should revert the commit from step 2 between steps 4 and 5.

Reply:
Yeah let me address this in the next PR